v3/bin/build

```diff
@@ -7,25 +7,47 @@ cd $(dirname "${BASH_SOURCE[0]}")/..
 stack=$1
 lifecycle_ref=$2
 samples_tgz=https://api.github.com/repos/buildpack/samples/tarball/master
-buildpack_dir=$(pwd)/images/detect/buildpacks
-nodejs_dir=$buildpack_dir/sh.packs.samples.buildpack.nodejs/latest
+buildpacks_dir=$(pwd)/images/detect/buildpacks
 
 if [[ -z $stack ]]; then
   >&2 echo "Stack not specified."
   exit 1
 fi
 
-rm -rf "$buildpack_dir"
-mkdir -p "$nodejs_dir"
+rm -rf "$buildpacks_dir"
+mkdir -p "$buildpacks_dir"
 
-tar_compat=$(if tar --version | grep -q "GNU tar" ; then echo '--wildcards'; fi)
-curl -sSL "$samples_tgz" | tar -xzf - -C "$nodejs_dir" --strip-components=2 $tar_compat 'buildpack-samples-*/nodejs-buildpack'
-
-version_dir=$(dirname "$nodejs_dir")/$(cat "$nodejs_dir/buildpack.toml" | yj -t | jq -r .version)
-mv "$nodejs_dir" "$version_dir"
-ln -s "$(basename "$version_dir")" "$nodejs_dir"
+if tar --version | grep -q "GNU tar"; then
+  curl -sSL "$samples_tgz" | tar -xzf - -C "$buildpacks_dir" --strip-components=1 --wildcards "buildpack-samples-*"
+else
+  curl -sSL "$samples_tgz" | tar -xzf - -C "$buildpacks_dir" --strip-components=1 "buildpack-samples-*"
+fi
 
-echo 'groups = [{ repository = "packs/v3", build-image = "packs/v3:build", run-image = "packs/v3:run", buildpacks = [{ id = "sh.packs.samples.buildpack.nodejs", version = "latest" }] }]' > "$buildpack_dir/order.toml"
+cat << TOML > "$buildpacks_dir/order.toml"
+[[groups]]
+  build-image = "packs/v3:build"
+  run-image = "packs/v3:run"
+TOML
+
+for buildpack in "nodejs" "java"; do
+  echo "--> Adding buildpack: ${buildpack}"
+  bp_dir=${buildpacks_dir}/sh.packs.samples.buildpack.${buildpack}
+  mkdir -p ${bp_dir}
+
+  latest_dir=${bp_dir}/latest
+  mv ${buildpacks_dir}/${buildpack}-buildpack ${latest_dir}
+
+  version_dir=$(dirname "$latest_dir")/$(cat "$latest_dir/buildpack.toml" | yj -t | jq -r .version)
+  mv "$latest_dir" "$version_dir"
+  ln -s "$(basename "$version_dir")" "$latest_dir"
+
+cat << TOML >> "$buildpacks_dir/order.toml"
+  [[groups.buildpacks]]
+    id = "sh.packs.samples.buildpack.${buildpack}"
+    version = "latest"
+    optional = true
```

Comment:
The `optional` field lets the group pass even if an individual optional buildpack fails. This is to permit extension buildpacks that may not apply. Essentially:
```
Group 1: newrelic-buildpack (optional) | java-buildpack
```
is a shorter, more efficient version of:
```
Group 1: newrelic-buildpack | java-buildpack
Group 2: java-buildpack
```
Making all buildpacks in a group optional means all buildpacks that individually pass detection would apply. Not sure if this is what we want. If Node.js + Java is common (for Java apps with Node assets), maybe something like:
```toml
[[groups]]
  build-image = "packs/v3:build"
  run-image = "packs/v3:run"
  buildpacks = [
    { id = "io.buildpacks.samples.buildpacks.nodejs, version = "latest", optional = true },
    { id = "io.buildpacks.samples.buildpacks.java, version = "latest" },
  ]
[[groups]]
  build-image = "packs/v3:build"
  run-image = "packs/v3:run"
  buildpacks = [
    { id = "io.buildpacks.samples.buildpacks.nodejs, version = "latest" },
  ]
```

Reply:
I actually think `optional` might be broken. When it's `false`, the whole detection fails when one buildpack fails.

Reply:
I started with the `order.toml` you have there, and my detect failed because the Node.js detect failed.

Reply:
wait no, my toml was different.